caikit_nlp/modules/text_generation/peft_config.py

```diff
@@ -0,0 +1,193 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Standard
+from enum import Enum
+import os
+
+# Third Party
+from peft import MultitaskPromptTuningInit
+from transformers import AutoConfig
+
+# First Party
+from caikit import get_config
+
+# Local
+from ...data_model import PromptOutputModelType
+from ...resources.pretrained_model import PretrainedModelBase
+from ...toolkit.data_type_utils import get_torch_dtype
+from ...toolkit.verbalizer_utils import is_valid_verbalizer
+
+# NOTE: We do not allow all the methods exposed by MPT / PT, such as `EXACT_SOURCE_TASK`
+# since those are for experimental use and would not be useful / applicable
+# for end-user use-cases
+allowed_tuning_init_methods = [
+    "TEXT",
+    "RANDOM",
+    "ONLY_SOURCE_SHARED",
+    "AVERAGE_SOURCE_TASKS",
+]
+
+
+class TuningType(str, Enum):
+    PROMPT_TUNING = "PROMPT_TUNING"
+    MULTITASK_PROMPT_TUNING = "MULTITASK_PROMPT_TUNING"
+    # MULTITASK_PREFIX_TUNING = "MULTITASK_PREFIX_TUNING"
+    # P_TUNING = "P_TUNING"
+    # PREFIX_TUNING = "PREFIX_TUNING"
+    # LORA = "LORA"
+
+
+def validate_peft_config(
+    tuning_type, tuning_config, error, log, base_model, cls, torch_dtype, verbalizer
```

Comment:
We usually initialize logger per module / file to allow easy backtracking.  To do this, we can initialize `log` at the beginning of this file and re-use that everywhere in this file.
```
log = alog.use_channel("PFT_CNFG_TLKT")
```